### PR TITLE
Feature/ab2d 747 filter out claims on dates

### PR DESCRIPTION
**Pull request by lailasharshar**
### Summary

Filter out explanation of benefit objects by dates subscribing and attestation date. I have it commented where it should go but still have questions about where the parameters should come from in the worker.

### Checklist

- [ ] Tests and linting pass
- [ ] Code checked for PHI/PII exposure

### Security
No security issues except to make sure the objects don't make it in that shouldn't

### CODE REVIEW

File: worker/src/main/java/gov/cms/ab2d/worker/adapter/bluebutton/PatientClaimsProcessorImpl.java [hunk 1]
```diff
@@ -154,6 +154,12 @@ private void tryLock(Lock lock) {
     }
 
     private List<Resource> extractResources(List<BundleEntryComponent> entries) {
+        /**
+         * To filter out by date: (with params List<Integer> validMonths, year, attestationDate
+         *
+         *    List<FilterOutByDate> dateRanges = FilterOutByDate.getDateRanges(validMonths, year);
+         *    List<ExplanationOfBenefit> benefits = FilterOutByDate.filterByDate(resources, attestationDate, dateRanges);
+         */
```
Comment (dkrylovsb) [inline worker/src/main/java/gov/cms/ab2d/worker/adapter/bluebutton/PatientClaimsProcessorImpl.java]: `PatientDTO.monthsUnderContract` will have the coverage months you need.

Comment (dkrylovsb) [reply worker/src/main/java/gov/cms/ab2d/worker/adapter/bluebutton/PatientClaimsProcessorImpl.java]: ...and `Contract.attestedOn` is the attestation date.

File: bfd/src/test/java/gov/cms/ab2d/bfd/client/BFDClientConfigurationTest.java [hunk 2]
```diff
@@ -1,38 +0,0 @@
-package gov.cms.ab2d.bfd.client;
```
Comment (dkrylovsb) [inline bfd/src/test/java/gov/cms/ab2d/bfd/client/BFDClientConfigurationTest.java]: I guess the deletion was unintentional?

Comment (lailasharshar) [reply bfd/src/test/java/gov/cms/ab2d/bfd/client/BFDClientConfigurationTest.java]: I have no memory of this

Comment (dkrylovsb) [reply bfd/src/test/java/gov/cms/ab2d/bfd/client/BFDClientConfigurationTest.java]: must have been accidental. Satheesh restored the file.

**Pull request merged**